packages/rest/src/controllers/proofs/ProofController.ts

```diff
@@ -0,0 +1,236 @@
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`)
+      }
+      throw new InternalServerError(`something went wrong: ${error}`)
+    }
+  }
+
+  /**
+   * Accept a presentation proposal as verifier (by sending a presentation request message) to the connection
+   * associated with the proof record.
+   */
+  @Post('/:proofRecordId/accept-proposal')
+  public async acceptProposal(
+    @Param('proofRecordId') proofRecordId: string,
+    @Body() proposal: AcceptProofProposalRequest
+  ) {
+    try {
+      const proof = await this.agent.proofs.acceptProposal(proofRecordId, proposal)
+
+      return proof.toJSON()
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`proof with proofRecordId "${proofRecordId}" not found.`)
+      }
+      throw new InternalServerError(`something went wrong: ${error}`)
+    }
+  }
+
+  /**
+   * Creates a presentation request not bound to any proposal or existing connection
+   */
+  @Post('/request-outofband-proof')
+  public async requestProofOutOfBand(
+    @Body()
+    request: ProofRequestRequest
+  ) {
+    const { proofRequest, comment } = request
+
+    const proof = await this.agent.proofs.createOutOfBandRequest(proofRequest, {
+      comment,
+    })
+
+    return {
+      message: `https://wwww.example.com/?c_i=${JsonEncoder.toBase64URL(proof.requestMessage.toJSON())}`,
+      proofRecord: proof.proofRecord,
+    }
+  }
+
+  /**
+   * Creates a presentation request bound to existing connection
+   */
+  @Post('/:connectionId/request-proof')
+  public async requestProof(
+    @Param('connectionId') connectionId: string,
+    @Body()
+    request: ProofRequestRequest
+  ) {
+    try {
+      const { proofRequest, comment } = request
+
+      const proof = await this.agent.proofs.requestProof(connectionId, proofRequest, {
+        comment,
+      })
+
+      return proof.toJSON()
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`)
+      }
+      throw new InternalServerError(`something went wrong: ${error}`)
+    }
+  }
+
+  /**
+   * Accept a presentation request as prover (by sending a presentation message) to the connection
+   * associated with the proof record.
+   */
+  @Post('/:proofRecordId/accept-request')
+  public async acceptRequest(
+    @Param('proofRecordId') proofRecordId: string,
+    @Body() request: PresentationProofRequestRequest
+  ) {
+    try {
+      const { proofRequest, presentationProposal, comment } = request
+
+      const retrievedCredentials = await this.agent.proofs.getRequestedCredentialsForProofRequest(
+        proofRequest,
+        presentationProposal
+      )
+
+      const requestedCredentials = this.agent.proofs.autoSelectCredentialsForProofRequest(retrievedCredentials)
```

Comment:
ACA-Py allows to select which credential to use for a proof request. I think it is fine to autmoatically select which credential to use for now, but could you create an issue for it?

Reply:
https://github.com/hyperledger/aries-framework-javascript-ext/issues/21